components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/AccountConfirmationValidationHandler.java

```diff
@@ -117,4 +123,24 @@ private boolean isUserAccountConfirmed(User user) throws IdentityEventException
         return userConfirmed ;
     }
 
+    private boolean isUserExistsInDomain(UserStoreManager userStoreManager, String userName)
+            throws IdentityEventException {
+
+        boolean isExists = false;
+        try {
+            if (userStoreManager.isExistingUser(userName)) {
+                isExists = true;
+            }
+        } catch (UserStoreException e) {
+            throw new IdentityEventException("Error occurred while check user existence: " + userName, e);
+        }
+        return isExists;
+    }
+
+    private boolean isAuthPolicyAccountExistCheck() {
+        String authpolicy = IdentityUtil.getProperty("AuthenticationPolicy.CheckAccountExist");
+        return authpolicy == null || Boolean.parseBoolean(IdentityUtil
```

Comment:
Doesn't parseBoolean handle null case ? Do we need to explicitly check it ?

Reply:
Boolean.parseBoolean(null) returns false but we want to return true when the config is not available.. That's why it handled specifically.

Reply:
got it . Then please avoid retrieving again from property map since we are already assigning it to a variable.